controllers/configurationpolicy_controller.go

```diff
@@ -717,11 +719,13 @@ func (r *ConfigurationPolicyReconciler) handleObjectTemplates(plc policyv1.Confi
 
 		// set finalizer if it hasn't been set
 		if !objHasFinalizer(&plc, pruneObjectFinalizer) {
-			plc.SetFinalizers(addObjFinalizer(&plc, pruneObjectFinalizer))
+			mergePatch := []byte(`{"metadata": {"finalizers": ["` + pruneObjectFinalizer + `"]}}`)
 
-			err := r.Update(context.TODO(), &plc)
+			err := r.Patch(context.TODO(), &plc, client.RawPatch(types.MergePatchType, mergePatch))
```

Comment:
@mprahl could this accidentally remove any other finalizer on the resource? I was expecting a JSON patch like `[{"op":"add","path":"/metadata/finalizers/-","value":"..."}]`

Reply:
Good catch. I thought the merge patch would act like strategic merge but I got the two confused. Here is a PR with the fix:
https://github.com/open-cluster-management-io/config-policy-controller/pull/119